wp-permastructure.php

```diff
@@ -235,13 +235,17 @@ public function add_permastructs( $rules ) {
 	 * custom taxonomies as well as the standard %author% etc...
 	 *
 	 * @param string $post_link The post URL
-	 * @param object $post      The post object
+	 * @param WP_Post $post      The post object
 	 * @param bool $leavename Passed to pre_post_link filter
 	 * @param bool $sample    Used in admin if generating an example permalink
 	 *
 	 * @return string    The parsed permalink
 	 */
-	public function parse_permalinks( $post_link, WP_Post $post, $leavename, $sample = false ) {
+	public function parse_permalinks( $post_link, $post, $leavename, $sample = false ) {
+		// Yoast Sitemap plug-in doesn't pass a WP_Post object causing a fatal, so we'll check for it and return.

```

Comment:
@mmaton what does Yoast sitemap pass here exactly? If it's a post ID then the first line of this method could be:

```php
$post = get_post( $post );
```

then it won't matter what the type is, object, int or null for global post.